### ISSUE TITLE
Better Windows support

### ISSUE THREAD

**Issue by LMP88959** [LMP88959/Digital-Subband-Video-2#8]
I don't plan on adding better Windows support nor do I have a good avenue to test Windows software so this issue is mainly for whoever is interested to work on.

The main issue is the lack of support for large input files due to 'long' being 32-bit on Windows.

This whole issue should amount to:
1. replacing 'long' with 'size_t' where appropriate in the file IO routines
2. (possibly) compiling with `-D__FILE_OFFSET_BITS=64` as @notcancername suggested

**Comment by kalmard0** [comment]
I discovered an additional Windows issue today, that prevents the `dsv2mp4` script from operating on Windows.

Windows (or MSVC's libraries?) treats `stdout` as text-mode and automatically converts `\n` to `\r\n` when writing to it. This means the yuv4mpeg2 stream output by the `dsv` command-line tool will ultimately contain incorrect values.

As far as I can tell, the only good solution (other than writing to a file in binary mode and then re-reading it for conversion) is to change this behavior using a vendor-specific function:

```
#ifdef _WIN32
#include <fcntl.h>
#include <io.h>
#endif

...

#ifdef _WIN32
  _setmode(_fileno(stdout), _O_BINARY);
#endif
```

**Comment by LMP88959** [comment]
Great find, thank you! So this _setmode call needs to happen essentially at the top of the main() function? Also, I'm curious if you've tried it to see if it fixes the issue?

**Comment by kalmard0** [comment]
> Interesting, I wonder if the ASCII parts of the Y4M format (e.g `"FRAME\n"`) cause issues as well if \r is added to it...

Good point! I guess in theory it is non-standard, but ffmpeg wasn't complaining about that part.

> Which flags? Or is it a loglevel issue?

What I found is that ffmpeg returns success (0) even on failure using the default mode, but the resulting video is trimmed at some point with occasional garbled data at the end. What works for me is `-err_detect aggressive -xerror` - but I don't know if this changes anything else, is too aggressive for your use, etc..

**Comment by LMP88959** [comment]
Very interesting indeed, maybe the byte for '\n' doesn't appear in the decoded YUV until a certain point. I would expect a total crash but I don't know how FFmpeg parses Y4M because at least in DSV I hop to the next frame because I know how many bytes each frame should be but if all instances of '\n' (1 byte) are getting converted to 2 bytes then I would expect a naive parser to crash very quickly.

**Comment by LMP88959** [comment]
Can you try this please and see if it fixes that issue? [Commit 7fe949cf](https://github.com/LMP88959/Digital-Subband-Video-2/commit/7fe949cf768294737c305f9609baa4be3261a44d)

**Comment by kalmard0** [comment]
> Can you try this please and see if it fixes that issue? [Commit 7fe949cf](https://github.com/LMP88959/Digital-Subband-Video-2/commit/7fe949cf768294737c305f9609baa4be3261a44d)

I put together the dsv cli-based converter setup again in my build system and indeed, before that commit it's broken, after that commit it's solved.
